src/main/java/org/venice/beachfront/bfapi/services/PiazzaJobPoller.java

```diff
@@ -0,0 +1,112 @@
+
+import javax.annotation.PostConstruct;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.venice.beachfront.bfapi.model.Job;
+import org.venice.beachfront.bfapi.model.exception.UserException;
+import org.venice.beachfront.bfapi.model.piazza.StatusMetadata;
+
+/**
+ * Scheduling class that manages the constant polling for Status of Shoreline detections Jobs in Piazza
+ */
+@Component
+public class PiazzaJobPoller {
+	@Value("${piazza.poll.frequency.seconds}")
+	private int POLL_FREQUENCY_SECONDS;
+
+	@Autowired
+	private JobService jobService;
+	@Autowired
+	private PiazzaService piazzaService;
+
+	private PollStatusTask pollStatusTask = new PollStatusTask();
+	private Timer pollTimer = new Timer();
+
+	/**
+	 * Begins scheduled polling of Piazza Job Status
+	 */
+	@PostConstruct
+	public void startPolling() {
+		pollTimer.schedule(pollStatusTask, 10000, POLL_FREQUENCY_SECONDS * (long) 1000);
+	}
+
+	public void stopPolling() {
+		pollTimer.cancel();
+	}
+
+	/**
+	 * Timer Task that will, on a schedule, poll for the Status of Piazza Jobs
+	 */
+	public class PollStatusTask extends TimerTask {
+		@Override
+		public void run() {
+			// Get the list of outstanding jobs that need to be queried
+			List<Job> outstandingJobs = jobService.getOutstandingJobs();
+			for (Job job : outstandingJobs) {
+				// For each job, query the status of that job in Piazza
+				StatusMetadata status = null;
+				try {
+					status = piazzaService.getJobStatus(job.getJobId());
+				} catch (UserException exception) {
+					// TODO - Get the age of the job and determine if a complete failure based on timeout is necessary
+					continue;
+				}
+				processJobStatus(job, status);
+			}
+		}
+
+		/**
+		 * Processes the returned Piazza Job Status for the specified Beachfront Job
+		 * 
+		 * @param job
+		 *            The Job to update
+		 * @param status
+		 *            The Status of the corresponding Piazza Job
+		 */
+		private void processJobStatus(Job job, StatusMetadata status) {
+			// Update the Status of the Job
+			job.setStatus(status.getStatus());
+			// Process based on the status
+			if (status.isStatusIncomplete()) {
+				// Nothing to do here. Polling will continue for this job.
+			} else if (status.isStatusSuccess()) {
+				// Get the Data ID of the Job and download the file bytes
+				try {
+					byte[] detectionBytes = piazzaService.downloadData(status.getDataId());
+					// TODO: John - how do I save these GeoJSON bytes to the database?
+				} catch (UserException exception) {
+					exception.printStackTrace();
+					// Fail the Job as we have failed to download the bytes
+					job.setStatus(Job.STATUS_ERROR);
+				}
+			} else if (status.isStatusError()) {
+				// https://github.com/venicegeo/bf-api/blob/master/bfapi/db/jobs.py#L125
+				// TODO: John, how do I save to the job_error table?
```

Comment:
Now that jobuser and detection are about to be merged, the joberror table is next on my list